src/fu-security-attr.c

```diff
@@ -203,6 +203,598 @@ fu_security_attr_get_name(FwupdSecurityAttr *attr)
+		/* TRANSLATORS: Title: attestation means verifying your system state */
+		return _("System Attestation");
+	}
+
+	if (g_strcmp0(appstream_id, FWUPD_SECURITY_ATTR_ID_MEI_MANUFACTURING_MODE) == 0 ||
+	    g_strcmp0(appstream_id, FWUPD_SECURITY_ATTR_ID_MEI_OVERRIDE_STRAP) == 0 ||
+	    g_strcmp0(appstream_id, FWUPD_SECURITY_ATTR_ID_MEI_VERSION) == 0 ||
+	    g_strcmp0(appstream_id, FWUPD_SECURITY_ATTR_ID_PLATFORM_DEBUG_ENABLED) == 0 ||
+	    g_strcmp0(appstream_id, FWUPD_SECURITY_ATTR_ID_PLATFORM_DEBUG_LOCKED) == 0 ||
+	    g_strcmp0(appstream_id, FWUPD_SECURITY_ATTR_ID_PLATFORM_FUSED) == 0) {
+		/* TRANSLATORS: Title: as in your entire computer */
+		return _("System Protection");
+	}
+	return NULL;
+}
+
+/* one line saying the problem, then one line saying why -- or how to fix the problem */
+gchar *
+fu_security_attr_get_description(FwupdSecurityAttr *attr)
+{
+	const gchar *appstream_id = fwupd_security_attr_get_appstream_id(attr);
+	g_autoptr(GPtrArray) strs = g_ptr_array_new();
+	gboolean contact_hw_vendor = FALSE;
+	gboolean check_bios_settings = FALSE;
+
+	if (g_strcmp0(appstream_id, FWUPD_SECURITY_ATTR_ID_SPI_BIOSWE) == 0 &&
+	    g_strcmp0(appstream_id, FWUPD_SECURITY_ATTR_ID_AMD_SPI_WRITE_PROTECTION) == 0) {
+		if (fwupd_security_attr_has_flag(attr, FWUPD_SECURITY_ATTR_FLAG_SUCCESS)) {
+			/* TRANSLATORS: longer success message (unlikely to see) */
+			g_ptr_array_add(strs, _("The system firmware is locked and not writable."));
+		} else {
+			g_ptr_array_add(
+			    strs,
+			    /* TRANSLATORS: longer failure message */
+			    _("The system firmware is unlocked and writable by any user."));
+			contact_hw_vendor = TRUE;
+		}
+	}
+	if (g_strcmp0(appstream_id, FWUPD_SECURITY_ATTR_ID_SPI_BLE) == 0) {
+		if (fwupd_security_attr_has_flag(attr, FWUPD_SECURITY_ATTR_FLAG_SUCCESS)) {
+			/* TRANSLATORS: longer success message (unlikely to see) */
+			g_ptr_array_add(strs, _("The system firmware cannot be easily unlocked."));
+		} else {
+			g_ptr_array_add(
+			    strs,
+			    /* TRANSLATORS: longer failure message */
+			    _("The system firmware can be unlocked and written by any user."));
+			contact_hw_vendor = TRUE;
+		}
+	}
+	if (g_strcmp0(appstream_id, FWUPD_SECURITY_ATTR_ID_SPI_SMM_BWP) == 0) {
+		if (fwupd_security_attr_has_flag(attr, FWUPD_SECURITY_ATTR_FLAG_SUCCESS)) {
+			/* TRANSLATORS: longer success message (unlikely to see) */
+			g_ptr_array_add(strs, _("The system firmware cannot be unlocked."));
+		} else {
+			g_ptr_array_add(
+			    strs,
+			    /* TRANSLATORS: longer failure message */
+			    _("The system firmware could be unlocked and written by any user."));
+			contact_hw_vendor = TRUE;
+		}
+	}
+	if (g_strcmp0(appstream_id, FWUPD_SECURITY_ATTR_ID_SPI_DESCRIPTOR) == 0) {
+		if (fwupd_security_attr_has_flag(attr, FWUPD_SECURITY_ATTR_FLAG_SUCCESS)) {
+			g_ptr_array_add(strs,
+					/* TRANSLATORS: longer success message (unlikely to see) */
+					_("Critical parts of the system firmware are protected."));
```

Comment:
what does critical part mean here?

Reply:
Critical areas? There are a few attacks where the descriptor is unlocked and you can change the protected ranges, and I wanted to signify that this allowed other bad things to happen.

Reply:
What about `An important area of the system firmware could be changed by a privileged user.`